Marlin/src/HAL/NATIVE_SIM/HAL.h

```diff
@@ -28,6 +28,8 @@
 #include <algorithm>
 #include "pinmapping.h"
 
+#define strlcpy(A, B, C)   strncpy(A, B, (C) - 1)  // Add strlcpy
```

Comment:
This won't guarantee NUL-termination, if `strlen(B) >= C`. You'll need to explicitly set `A[C-1] = '\0'`;

Reply:
I just replicated https://github.com/MarlinFirmware/Marlin/pull/26513/files#diff-1b2fb6d5dec43976593c1ba353f8de014cc09249d8a3880abf331e323bc778c0